core/src/staked_nodes_updater_service.rs

```diff
@@ -19,21 +20,55 @@ pub struct StakedNodesUpdaterService {
     thread_hdl: JoinHandle<()>,
 }
 
+fn firedancer_write_stake(
+    mvcc: &mut Mvcc,
```

Comment:
will assume taking an &mut here and the Mvcc rust interface respects rust's rules...